kafka/configuration.go

```diff
@@ -44,8 +54,21 @@ type BrokerConfiguration struct {
 	Enabled          bool          `mapstructure:"enabled" toml:"enabled"`
 }
 
+// MultiBrokerConfiguration represents configuration of Kafka broker with
+// multiple instances running on different hosts
+type MultiBrokerConfiguration struct {
+	Addresses        []string            `mapstructure:"addresses" toml:"addresses"`
+	SecurityProtocol string              `mapstructure:"security_protocol" toml:"security_protocol"`
+	SASLConfigs      []SASLConfiguration `mapstructure:"sasl_configs" toml:"sasl_configs"`
+	Topic            string              `mapstructure:"topic" toml:"topic"`
+	Timeout          time.Duration       `mapstructure:"timeout" toml:"timeout"`
+	Group            string              `mapstructure:"group" toml:"group"`
+	ClientID         string              `mapstructure:"client_id" toml:"client_id"`
+	Enabled          bool                `mapstructure:"enabled" toml:"enabled"`
+}
+
```

Comment:
The whole structure can be an array of "SingleBrokerConfiguration"

Reply:
It has a way more consequences on the services that would use it (more files to change, more tests to update, etc.)

I'm testing the changes in inutile using the ccx-notification-writer for example, and changes are needed everywhere . Be prepared :D

<img width="617" alt="image" src="https://github.com/RedHatInsights/insights-operator-utils/assets/15891027/3c2f8f59-ef06-4a08-8058-603a111e7394">

Reply:
But I'm still doing it, as it seems like a cleaner approach long term